File: benchmarks/concurrency/actor_spawn.go
package main

import (
	"fmt"
	"sync"
)

func main() {
	var wg sync.WaitGroup
	for i := 0; i < 10000; i++ {
		wg.Add(1)
		go func(id int) {
			_ = id
			wg.Done()
		}(i)
	}
	wg.Wait()
	fmt.Println("Spawned 10000 actors.")
}

File: benchmarks/concurrency/actor_spawn.rs
use std::sync::mpsc;
use std::thread;

fn main() {
    let mut handles = Vec::new();
    for i in 0..10000 {
        handles.push(thread::spawn(move || {
            let _id = i;
        }));
    }
    for h in handles {
        h.join().unwrap();
    }
    println!("Spawned 10000 actors.");
}
